lab6/src/components/Header.jsx

```diff
@@ -0,0 +1,47 @@
+import '../css/base.css'
+import SideNav from './SideNav';
+import { ReactComponent as MenuIconS } from '../svgs/MenuIconS.svg'
+import { Link } from "react-router-dom"
+import { useState } from 'react';
+
+const Header = () => {
+  const [isShown, setIsShown] = useState(false)
+  const showHideSideNav = () => setIsShown(!isShown)
+
+  return (
+    <div className="header-container">
+      <div className="header">
+        <div className="side-nav-counter-block"></div>
+        <nav>
+            <ul className="header-ul header-ul-left">
+                <li>
+                    <Link to="/" className="a text nav-text">About</Link>
+                </li>
+                <li>
+                    <Link to="/" className="a text nav-text">Support</Link>
+                </li>
+            </ul>
+        </nav>
+        <Link to="/" className="a text logo-text cursor-pointer">Spotifee</Link>
+        <nav>
+            <ul className="header-ul header-ul-right">
+                <li>
+                    <Link to="/SignIn" className="a text nav-text cursor-pointer">Sign In</Link>
+                </li>
+                <li>
+                    <a href="https://www.spotify.com/us/signup" className="a text nav-text">Sign Up</a>
+                </li>
+            </ul>
+        </nav>
+        <button 
+          className={isShown ? "menu-button menu-button-media cursor-pointer menu-button-selected" : "menu-button menu-button-media cursor-pointer"}
+          onClick={showHideSideNav}>
+          <MenuIconS />
+        </button>
```

Comment:
Should be a separate component. with class `button`, and props, which is gonna decide how the classname gonna looks like 

```
interface IProps {
  className?: string;
  onClick?: React.MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
  autoFocus?: boolean;
  value?: string;
  disabled?: boolean;
  type?: 'normal' | 'accent';
  to?: string;
}

class Button extends React.Component<IProps> {

  render() {
    const { onClick, className, children, autoFocus, value, disabled, type, to } = this.props;

    return (
      <>
        {to
        ? <Link
            to={to}
            className={clsx('button', {
              'button--normal': !type || type === 'normal',
              'button--accent': type === 'accent',
              'button--disabled': disabled,
            }, className)}
            onClick={onClick}
          >
            {children}
          </Link>
        : <button
            autoFocus={autoFocus}
            className={clsx('button', {
              'button--normal': !type || type === 'normal',
              'button--accent': type === 'accent',
              'button--disabled': disabled,
            }, className)}
            onClick={onClick}
            value={value}
            disabled={disabled}
          >
            {children}
          </button>
        }
      </>
    )
  }
}

export default Button;
```

Smth like this.